app/src/main/java/org/jellyfin/mobile/MainActivity.kt

```diff
@@ -53,12 +54,19 @@ class MainActivity : AppCompatActivity() {
         bindService(Intent(this, RemotePlayerService::class.java), serviceConnection, Service.BIND_AUTO_CREATE)
 
         // Load UI
-        appPreferences.instanceUrl?.toHttpUrlOrNull().also { url ->
-            with(supportFragmentManager) {
-                if (url != null) {
-                    replaceFragment<WebViewFragment>()
-                } else {
-                    replaceFragment<ConnectFragment>()
+        lifecycleScope.launch {
+            mainViewModel.serverState.collect { state ->
+                with(supportFragmentManager) {
+                    when (state) {
+                        ServerState.Pending -> {
+                            // TODO add loading indicator
+                        }
```

Comment:
This can be ignored for now.

Reply:
Nope, compose handles it differently. But I haven't decided how to do it yet. The branch is meant mostly to have the when exhaustive.